census/views.py

```diff
@@ -60,97 +61,159 @@ def events_of(size=1000):
+                                             "%Y-%m-%d %H:%M:%S")
+
+                filter_args['start_date'] = start_date
+                filter_args['end_date'] = end_date
+        else:
+            # If no payload is passed to the request, simply fetch current and
+            # future approved events
+            filter_args['start_date'] = datetime.now().replace(hour=0, minute=0, second=0)
+            filter_args['end_date'] = datetime.now() + timedelta(days=7)
+        return self.fetch_events_from_db(**filter_args)
+
+    def get_valid_params(self, query_params):
+        """
+        Validate URL Query parameters and convert them to Python friendly values where applicable.
+        Invalid query parameters/values will be ignored.
+
+        :param query_params: Dictionary containing query parameters and corresponding values.
+        :return: Dictionary containing only valid parameters
+        """
+        valid_params = {}
+        for param in query_params:
+            if param == 'day' and query_params['day'].isdigit():
+                valid_params['day'] = query_params['day']
+            elif param == 'month' and query_params['month'].isdigit():
+                valid_params['month'] = query_params['month']
+            elif param == 'year' and query_params['year'].isdigit():
+                valid_params['year'] = query_params['year']
+            elif param == 'search' \
+                    and query_params['search'] \
+                    and not query_params['search'].strip() == "":
+                valid_params['search'] = query_params['search']
+            elif param == 'isMonthly':
+                if query_params['isMonthly'] == 'true':
+                    valid_params['isMonthly'] = True
+                elif query_params['isMonthly'] == 'false':
+                    valid_params['isMonthly'] = False
+        return valid_params
+
+    def fetch_events_from_db(self, **kwargs):
+        """
+        Retrieve matching approved events data.
+
+        :param kwargs: Arguments containing DB fields and values
+        :return: Query set of matching events
+        """
+
+        start_date = kwargs.get('start_date')
+        end_date = kwargs.get('end_date')
+        month = kwargs.get('month')
+        year = kwargs.get('year')
+        search = kwargs.get('search')
+
+        if bool(start_date) ^ bool(end_date):
+            # If start date is provided but not end date, or vice-versa, raise error
+            raise Exception
+        query = Q()
+        if month:
+            query = query & Q(start_datetime__month=month, start_datetime__year=year)
+        if start_date and end_date:
+            query = query & Q(start_datetime__range=(current_timezone.localize(start_date),
+                                                     current_timezone.localize(end_date)))
+        if search:
+            query = query & (Q(title__icontains=search) | Q(description__icontains=search))
```

Comment:
As per the docs, it seems that we need to be using PostgreSQL DB before being able to use Django's full text search (https://docs.djangoproject.com/en/2.2/ref/contrib/postgres/search/)
So I thought I'll update this later when we migrate to Postgres. Currently, the search only matches title and description, but we would have to add more fields like location soon (https://github.com/openoakland/dos-census-events/issues/103)